### PR TITLE
Adds more gun type codexes and description changes

### PR DESCRIPTION
**Pull request by XSlayer300**
## About The Pull Request
Adds the 'flame' and 'explosive' weapon codexes that are missing from the said types of guns, this PR also expands upon the lore of 'energy' weapons as well, including other corrections.

In addition, this changes some gun descriptions to add a bit more fluff. And also weapon names as some of them are inconsistent.

## Why It's Good For The Game
Because we need more LORE.

## Changelog
:cl:
add: Added "explosive" and "flame" weapon type codexes and applied to their respective weapons.
tweak: Changed most of the gun descriptions, even a few names.
tweak: Changed the recent projectile names.
/:cl:

### CODE REVIEW

**Comment by Rohesie** [comment]
@grinkgo

File: code/game/objects/items/weapons/blades.dm [hunk 1]
```diff
@@ -71,7 +71,7 @@
 	return ..()
 
 /obj/item/weapon/combat_knife
-	name = "\improper M5 'Night Raider' survival knife"
+	name = "\improper M5 \"Night Raider\" survival knife"
```
Comment (spookydonut) [inline code/game/objects/items/weapons/blades.dm]: why

Comment (XSlayer300) [reply code/game/objects/items/weapons/blades.dm]: Because there are no quotation marks instead of apostrophe? Also, I need the slashes for the quotation marks because it'll break without them.

Comment (XSlayer300) [reply code/game/objects/items/weapons/blades.dm]: Anyways, reverted now.

File: code/modules/projectiles/updated_projectiles/guns/pistols.dm [hunk 2]
```diff
@@ -49,8 +49,8 @@
 //M4A3 PISTOL
 
 /obj/item/weapon/gun/pistol/m4a3
-	name = "\improper M4A3 service pistol"
-	desc = "An M4A3 Colt Service Pistol, the standard issue sidearm of the TerraGov Marine Corps. Uses 9mm pistol rounds."
+	name = "\improper T-27 pistol"
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/guns/pistols.dm]: Keep weapons like this one, which will be resprited, with their old name and description.
The idea is to preserve the old ones and add new ones.

File: code/modules/projectiles/updated_projectiles/guns/rifles.dm [hunk 3]
```diff
@@ -184,7 +184,7 @@
 
 /obj/item/weapon/gun/rifle/m41a1
 	name = "\improper M41A1 pulse rifle"
-	desc = "The standard issue rifle of the TerraGov Marine Corps. Commonly carried by most combat personnel. Uses 10x24mm caseless ammunition."
+	desc = "The standard issue rifle of the TerraGov Marine Corps, commonly carried by most combat personnel. However, due to its availability and resources, the M41A1 is a very rare sight on TerraGov systems. Uses 10x24mm caseless ammunition."
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/guns/rifles.dm]: No longer standard issue.

File: code/modules/projectiles/updated_projectiles/guns/pistols.dm [hunk 4]
```diff
@@ -456,8 +456,8 @@ It is a modified Beretta 93R, and can fire three round burst or single fire. Whe
 
 
 /obj/item/weapon/gun/pistol/knife
-	name = "\improper Knife"
-	desc = "There are a surprisingly large amount of attachment points on this.... knife."
+	name = "\improper M6 'Eclipse Raider' ballistic knife"
+	desc = "The back issue survival knife issued to TerraGov Marine Corps soldiers. There are a surprisingly large amount of attachment points on this.... knife."
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/guns/pistols.dm]:
```suggestion
	desc = "The back issue survival knife issued to TerraGov Marine Corps soldiers. There are a surprisingly large amount of attachment points on this... knife."
```

File: code/modules/projectiles/updated_projectiles/guns/shotguns.dm [hunk 5]
```diff
@@ -703,9 +714,4 @@ can cause issues with ammo types getting mixed up during the burst.
 
 	fire_delay = 6
 	accuracy_mult = 1.4
-	accuracy_mult_unwielded = 0.7
-	scatter = 15
-	scatter_unwielded = 40
-	recoil = 2
-	recoil_unwielded = 4
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/guns/shotguns.dm]: Are these changes intended?

Comment (XSlayer300) [reply code/modules/projectiles/updated_projectiles/guns/shotguns.dm]: Since the MBX share the same parent as the lever action, with the stats above heavily derived from it, I think it's not needed to copy the same thing and put it on the child weapon (the MBX).

**Comment by XSlayer300** [comment]
Besides that, I'm trying to make the 16 gauge shells (for the SX-16) use shotgun shell sprites (obtained from getting the bullets out from the mag) instead of generic bullet sprites. It did not end up too well though, now the sprites are invisible.

**Comment by Rohesie** [comment]
Let me know when that is fixed.

**Comment by XSlayer300** [comment]
I can't seem to do it sadly, limitations for me.

**Comment by Rohesie** [comment]
Revert those changes then?

File: code/modules/codex/entries/guns_codex.dm [hunk 6]
```diff
@@ -129,19 +129,49 @@
...
 		shotguns or some rifles, by opening the breech and clicking it with individual rounds. \
 		<br>"
 	lore_text = "Ballistic weapons are still used even now due to the relative expense of decent laser \
 		weapons, difficulties in maintaining them, and the sheer stopping and wounding power of solid slugs or \
 		composite shot. Using a ballistic weapon on a spacebound habitat is usually considered a serious undertaking, \
 		as a missed shot or careless use of automatic fire could rip open the hull or injure bystanders with ease."
+
+/datum/codex_entry/flame_weapons
+	display_name = "flame weapons"
+	mechanics_text = "This weapon is a flame weapon; it unleashes a stream of fire using a incinerator tank. You can \
```
Comment (Rohesie) [inline code/modules/codex/entries/guns_codex.dm]:
```suggestion
	mechanics_text = "This is a flame weapon; it unleashes a stream of fire using a incinerator tank. You can \
```

File: code/modules/codex/entries/guns_codex.dm [hunk 7]
```diff
@@ -129,19 +129,49 @@
...
+		refill liquid fuel by either using specialized fueltank backpacks made for the weapon or other fueltanks \
+		used by welders, although they're not effective. Unlike energy or ballistic weapons, flame weapons can \
```
Comment (Rohesie) [inline code/modules/codex/entries/guns_codex.dm]:
```suggestion
		used by welders, although they're not as effective. Unlike energy or ballistic weapons, flame weapons can \
```

File: code/modules/codex/entries/guns_codex.dm [hunk 8]
```diff
@@ -129,19 +129,49 @@
...
+		pierce through entities, people or not. Depending on the flame type, the flames' longevity, range and spray \
+		shape varies from type to type. \
+		<br>"
+	lore_text = "Traditional flamethrowers were developed in the 1st century and saw frequent use in the 20th century \
+		in the first two world wars on Terra. Thanks to their questionable purpose of burning people alive, rebel \
+		armies, militia groups and terrorist organizations make use of the weapon for centuries to come. It wasn't issued \
+		to inventory for the TerraGov Marine Corps along with lasguns until 2414, where aliens are vulnerable to intense heat."
```
Comment (Rohesie) [inline code/modules/codex/entries/guns_codex.dm]:
```suggestion
		to inventory for the TerraGov Marine Corps along with lasguns until 2414, due to the xenomorph threat and its vulnerability to intense heat."
```

File: code/modules/codex/entries/guns_codex.dm [hunk 9]
```diff
@@ -129,19 +129,49 @@
...
+		shape varies from type to type. \
+		<br>"
+	lore_text = "Traditional flamethrowers were developed in the 1st century and saw frequent use in the 20th century \
+		in the first two world wars on Terra. Thanks to their questionable purpose of burning people alive, rebel \
+		armies, militia groups and terrorist organizations make use of the weapon for centuries to come. It wasn't issued \
+		to inventory for the TerraGov Marine Corps along with lasguns until 2414, due to the xenomorph threat and its vulnerability to intense heat."
+
+/datum/codex_entry/explosive_weapons
+	display_name = "explosive weapons"
+	mechanics_text = "This weapon is an explosive weapon; it fires a projectile capable of detonating once it reaches its target. You can \
```
Comment (Rohesie) [inline code/modules/codex/entries/guns_codex.dm]:
```suggestion
	mechanics_text = "This is an explosive weapon; it fires a projectile capable of detonating once it reaches its target. You can \
```

File: code/modules/codex/entries/guns_codex.dm [hunk 10]
```diff
@@ -129,19 +129,49 @@
...
+		reload it by clicking the weapon with either rockets or grenades, depending on the weapon type. Explosive \
```
Comment (Rohesie) [inline code/modules/codex/entries/guns_codex.dm]:
```suggestion
		reload it by clicking the weapon with either rockets or grenades, depending on its type. Explosive \
```
